src/Launcher.vala

```diff
@@ -336,6 +342,43 @@ public class Dock.Launcher : Gtk.Box {
         }
     }
 
+    private void close_all_instances(){
+        if (app.running) {
+            if (app.action_group.lookup_action(CLOSE_WINDOWS_ACTION) != null) {
+                return;
+            }
+            
+            var close_item = new GLib.MenuItem(_(CLOSE_WINDOWS_LABEL), ACTION_PREFIX + CLOSE_WINDOWS_ACTION);
+            app.menu_model.append_item(close_item);
+    
+            var close_action = new SimpleAction(CLOSE_WINDOWS_ACTION, null);
```

Comment:
I think the close action should be installed during construction and stay installed.